sbndqm/Decode/PMT/ICARUS/DaqDecoderIcarusPMT_module.cc

```diff
@@ -207,95 +206,219 @@ void daq::DaqDecoderIcarusPMT::processFragment( const artdaq::Fragment &artdaqFr
   sbndaq::CAENV1730Event evt = *fragment.Event();
   sbndaq::CAENV1730EventHeader header = evt.Header;
   size_t nChannelsPerBoard = metafrag.nChannels;
-
   
   uint32_t ev_size_quad_bytes         = header.eventSize;
   uint32_t evt_header_size_quad_bytes = sizeof(sbndaq::CAENV1730EventHeader)/sizeof(uint32_t);
   uint32_t data_size_double_bytes     = 2*(ev_size_quad_bytes - evt_header_size_quad_bytes);
   uint32_t nSamplesPerChannel         = data_size_double_bytes/nChannelsPerBoard;
   uint16_t const enabledChannels      = header.ChannelMask();
 
+  // The fragment timestamp is computed in the DAQ using the server time and the TTT
+  // It is pointing at the END of the saved buffer, and in ns
   artdaq::Fragment::timestamp_t const fragmentTimestamp = artdaqFragment.timestamp(); 
   unsigned int const time_tag =  header.triggerTimeTag;
 
   auto const [ chDataMap, nEnabledChannels ] = setBitIndices<16U>(enabledChannels);
   const uint16_t* data_begin = reinterpret_cast<const uint16_t*>(artdaqFragment.dataBeginBytes() + sizeof(sbndaq::CAENV1730EventHeader));
 
+  // Channel waveform + channel temperatures
   std::vector<uint16_t> wvfm(nSamplesPerChannel);
-  float temperature = 0;
-
+  std::vector<float> temperatures;
 
   for( size_t digitizerChannel=0; digitizerChannel<nChannelsPerBoard; digitizerChannel++ ) {
 
+    // We don't have access to the offline DB: this is a different channel mapping
+    // Also offline DB doesn't deal with special channels (16th) 
```

Comment:
```
    // Also offline DB doesn't deal with special channels (16th) 
```
If this is a problem, we can also unilaterally solve it by adding those digitiser channels to the database. It needs a lot of care, but it can be done.

Reply:
That would probably the best long-term solution, including for the offiline where the only way to extract mapping information on the special channels is to parse the V1730 setup FHiCL.